### DISCUSSION TITLE
Custom Authentication

### DISCUSSION THREAD

**Discussion by ibrooz94** [fastapi-users/fastapi-users#1406] [Q&A] [answered]
I want to be able to get a users IP location from their request data and use that to compare with "safe_locations" stored in the User table. What piece of code from fastapi-users should I override.
I already tried authenticate in UserManager but I don't have access to the user's request object there.

Any help and pointers are appreciated. Thank You

**Comment by hgalytoby** [comment]
Which routes would you like to have this feature in?

**Reply by ibrooz94** [reply]
My login route preferably

**Reply by hgalytoby** [reply]
Can you explain what will be done after comparing "safe_locations"?

Such as not allowing logins to such things or...?

**Reply by ibrooz94** [reply]
A successful login, I'm currently using a CookieTransport so a cookie with a jwt token to access the API is set.

**Reply by ibrooz94** [reply]
The location data being recognized before login is mandatory, all other access controls mechanism are done after

**Reply by ibrooz94** [reply]
I have a feeling the login function on get_auth_router is what I need, but I'm having trouble on how to inherit that so i can override it.

**Reply by hgalytoby** [reply, marked as answer]
https://fastapi-users.github.io/fastapi-users/latest/configuration/user-manager/#on_after_login

Can't on_after_login be used?

**Reply by ibrooz94** [reply]
This seems viable, but i'll have to logout the user somehow or set a flag in the user to indicate the lack of permissions.

I'll look into this, Thank you. 
Is there anyway to override the login of get_auth_router, it suits my flow better
